tests\drivers\build_all\ethernet build fails on s32z270dc2

**Describe the bug**
tests\drivers\build_all\ethernet build fails on s32z270dc2

**To Reproduce**
`west build -p -b s32z2xxdc2/s32z270/rtu0 tests\drivers\build_all\ethernet`

**Expected behavior**
The test is built successful.

**Impact**
This may break CI causes PR cannot be merged

**Logs and console output**
```
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c: In function 'z_vrfy_mbox_send':
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:18:33: error: 'channel' undeclared (first use in this function); did you mean 'channel_id'?
   18 |         return z_impl_mbox_send(channel, msg);
      |                                 ^~~~~~~
      |                                 channel_id
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:18:33: note: each undeclared identifier is reported only once for each function it appears in
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:18:42: warning: passing argument 2 of 'z_impl_mbox_send' makes integer from pointer without a cast [-Wint-conversion]
   18 |         return z_impl_mbox_send(channel, msg);
      |                                          ^~~
      |                                          |
      |                                          const struct mbox_msg *
In file included from D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:8:
D:/upstream/zephyr/include/zephyr/drivers/mbox.h:265:54: note: expected 'mbox_channel_id_t' {aka 'unsigned int'} but argument is of type 'const struct mbox_msg *'
  265 |                                    mbox_channel_id_t channel_id,
      |                                    ~~~~~~~~~~~~~~~~~~^~~~~~~~~~
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:18:16: error: too few arguments to function 'z_impl_mbox_send'
   18 |         return z_impl_mbox_send(channel, msg);
      |                ^~~~~~~~~~~~~~~~
D:/upstream/zephyr/include/zephyr/drivers/mbox.h:264:19: note: declared here
  264 | static inline int z_impl_mbox_send(const struct device *dev,
      |                   ^~~~~~~~~~~~~~~~
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c: In function 'z_vrfy_mbox_set_enabled':
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:44:40: warning: passing argument 1 of 'z_impl_mbox_set_enabled' makes pointer from integer without a cast [-Wint-conversion]
   44 |         return z_impl_mbox_set_enabled(channel_id, enabled);
      |                                        ^~~~~~~~~~
      |                                        |
      |                                        mbox_channel_id_t {aka unsigned int}
D:/upstream/zephyr/include/zephyr/drivers/mbox.h:416:64: note: expected 'const struct device *' but argument is of type 'mbox_channel_id_t' {aka 'unsigned int'}
  416 | static inline int z_impl_mbox_set_enabled(const struct device *dev,
      |                                           ~~~~~~~~~~~~~~~~~~~~~^~~
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:44:16: error: too few arguments to function 'z_impl_mbox_set_enabled'
   44 |         return z_impl_mbox_set_enabled(channel_id, enabled);
      |                ^~~~~~~~~~~~~~~~~~~~~~~
D:/upstream/zephyr/include/zephyr/drivers/mbox.h:416:19: note: declared here
  416 | static inline int z_impl_mbox_set_enabled(const struct device *dev,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c: In function 'z_vrfy_mbox_send':
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:19:1: warning: control reaches end of non-void function [-Wreturn-type]
   19 | }
      | ^
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c: In function 'z_vrfy_mbox_set_enabled':
D:/upstream/zephyr/drivers/mbox/mbox_handlers.c:45:1: warning: control reaches end of non-void function [-Wreturn-type]
   45 | }
      | ^
[161/230] Building C object zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_nxp_s32_netc.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'D:\upstream\zephyr\build'
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
This is a regression of https://github.com/zephyrproject-rtos/zephyr/pull/69390

--- a/drivers/mbox/mbox_handlers.c
+++ b/drivers/mbox/mbox_handlers.c
@@ -15,7 +15,7 @@ static inline int z_vrfy_mbox_send(const struct device *dev,
 	K_OOPS(K_SYSCALL_MEMORY_READ(msg, sizeof(struct mbox_msg)));
 	K_OOPS(K_SYSCALL_MEMORY_READ(msg->data, msg->size));
 
-	return z_impl_mbox_send(channel, msg);
+	return z_impl_mbox_send(dev, channel_id, msg);
 }
 #include <syscalls/mbox_send_mrsh.c>
 
@@ -41,6 +41,6 @@ static inline int z_vrfy_mbox_set_enabled(const struct device *dev,
 {
 	K_OOPS(K_SYSCALL_DRIVER_MBOX(dev, set_enabled));
 
-	return z_impl_mbox_set_enabled(channel_id, enabled);
+	return z_impl_mbox_set_enabled(dev, channel_id, enabled);
 }
 #include <syscalls/mbox_set_enabled_mrsh.c>

